### ISSUE TITLE
file.css(.scss).erb causes `File to import not found or unreadable`

### ISSUE THREAD

**Issue by JeanMertz** [rails/sass-rails#139]
Using the ERB handler breaks reading/importing the files…

This works:

``` scss
// application.css.scss:
@import 'tester'

// tester.css(.scss)
body { color: #ABC; }
```

This does not:

``` scss
// application.css.scss:
@import 'tester'

// tester.css(.scss).erb
body { color: #ABC; }
```

**Sass::SyntaxError: File to import not found or unreadable**

At first I thought it was related to #119 or #94, but those fixes don't solve my problem. I do use `Rails 4.0.0.beta1` (together with `sass-rails 4.0.0.beta1`), so perhaps this could be related?

**Comment by lichtamberg** [comment]
I also have this issue with rails 3.2.22.1. Imports to .scss.erb are found, but not rendered with erb. I tried it with the 'sass_rails_patch' gem/path, but doesn't work. Any hints maybe?

**Comment by lichtamberg** [comment]
Lol, having the same problem again after 1,5 years...

**Comment by blaszczakphoto** [comment]
I;m still having the same issue. sass-rails (5.0.6).  "rails", "~> 5.1.0.beta1"

**Comment by jrochkind** [comment]
Yep appears to have been a regression at some point. 

`ActionView::Template::Error (File to import not found or unreadable: /Users/jrochkind/code/chf-sufia/app/assets/stylesheets/local/chf_image_viewer.scss.erb.`

Rails 5.0.6, sass-rails 5.0.6.

**Comment by srghma** [comment]
same issue, dump https://pastebin.com/iHxpZvzt

**Comment by rtsinani** [comment]
It looks like `.scss.erb` file doesn't work in subfolders, moving it to the `assets/stylesheets` folder directly worked for me.

**Comment by elialbert** [comment]
a year later and I'm experiencing the same problem - does .scss.erb still not work in subfolders with sprockets 4 manifest? or is my setup wrong somehow?

**Comment by jrochkind** [comment]
@elialbert I *think* the problem is that the `./app/assets/config/manifest.js` is set up to try including all of your .scss files as *top level* CSS files, while some of them are only meant to be included in other SCSS, not as top-level. So you can edit the `manifest.js` to instead include only those .scss you mean to be top-level. 

A confusing thing is that `manifest.js` has been in Rails apps for a while, but only starts being *used* once you upgrade to Sprockets 4, due to some conditional logic in sass-rails. Before Sprockets 4, it is ignored. 

I tried to write up what I could figure out about what's going on here: https://bibwild.wordpress.com/2019/10/15/sprockets-4-and-your-rails-app/

It is all very confusing; the greatly extended timeframe for the sprockets 3->4 upgrade led to different developers doing different things at different times, not all of which play well together, which became activated only when sprockets 4 was actually released.

**Comment by elialbert** [comment]
haha yeah thanks so much! I will read your blogpost but like, downgrading back to 3.7.2 is working just fine for right now so that's probably where we'll land.
thousands of rails devs all sitting on sprockets 3 waiting for someone else to fix the last things in the upgrade path...

**Comment by jrochkind** [comment]
I don't believe the ~sprockets and sprockets-rails~  maintainers (actually it's up to the Rails maintainers, see https://github.com/rails/rails/pull/28430) currently consider it broken. (I am not a committer/maintainer on any of those projects). 

So there is no point in "waiting for someone else to fix" it; that is not going to happen (unless you can change their minds).  You just need to figure out the right way to use sprockets 4 with rails as it is. 

Or, as you say, keep using sprockets 3 instead, that is certainly an option. 

It is true that upgrading from sprockets 3 to 4 may require some changes in your app to work -- either in the manifest.js, or in where you put your .sccs files on disk.

**Comment by elialbert** [comment]
that was a good blog post

**Comment by elialbert** [comment]
@jrochkind I just got back to upgrading to sprockets 4 on the same codebase I was on 3 years ago, and ran into a similar error, and searched google, and found my own conversation with you here, and had a momentary loss of sanity.

But to update you on your hypothesis with what was wrong last time - yes I did need to modify manifest.js slightly as per your blogpost; that was no problem. My bigger, weirder issue, which I just solved, was that sprockets running _in test under capybara only_ did not like my `app/stylesheets/x.scss` files to have a `.scss.erb` extension, even tho some of those files imported `@import "./config/x"` where x is `typography` but the actual file is `config/typography.scss.erb`. So I was getting the exact error in this thread, _only under test_, until I changed the top level stylesheets files to not use erb, _even tho they still import some erb files_.

very weird.
